src/ui/views/session/create-world/SceneUpload.tsx

```diff
@@ -0,0 +1,71 @@
+import { useEffect, useState, useCallback } from "react";
+import { IBlobHandle } from "@thirdroom/hydrogen-view-sdk";
+
+import { useHydrogen } from "../../../hooks/useHydrogen";
+import { useAttachmentUpload } from "../../../hooks/useAttachmentUpload";
+import { Label } from "../../../atoms/text/Label";
+import { Text } from "../../../atoms/text/Text";
+import { Button } from "../../../atoms/button/Button";
+import { Icon } from "../../../atoms/icon/Icon";
+import { SettingTile } from "../../components/setting-tile/SettingTile";
+import { bytesToSize } from "../../../utils/common";
+import { useDebounce } from "../../../hooks/useDebounce";
+import UploadIC from "../../../../../res/ic/upload.svg";
+
+export function SceneUpload({ onMxcChange }: { onMxcChange: (Url?: string) => void }) {
+  const { session, platform } = useHydrogen(true);
+  const [sceneBlob, setSceneBlob] = useState<IBlobHandle>();
+  const [sentBytes, setProgress] = useState(0);
+  const handleProgress = useCallback((sentBytes: number) => {
+    setProgress(sentBytes);
+  }, []);
+  const {
+    mxc,
+    upload,
+    cancel: cancelUpload,
+  } = useAttachmentUpload(session._hsApi, platform, useDebounce(handleProgress, { wait: 200, immediate: true }));
+
+  useEffect(() => {
+    setProgress(0);
+    if (sceneBlob) upload(sceneBlob);
+    else cancelUpload();
+  }, [sceneBlob, upload, cancelUpload]);
+
+  useEffect(() => onMxcChange(mxc), [mxc, onMxcChange]);
+
+  return (
+    <SettingTile className="grow basis-0" label={<Label>Scene *</Label>}>
+      <div className="flex flex-column items-start gap-xxs">
+        {sceneBlob ? (
+          <>
+            <Text variant="b3" weight="medium">
+              File: {sceneBlob.nativeBlob.name}
+            </Text>
+            <Text weight={mxc ? "regular" : "bold"} variant="b3">
+              {mxc && `Size: ${bytesToSize(sceneBlob.size)}`}
+              {!mxc && `Uploading: ${bytesToSize(sentBytes)} / ${bytesToSize(sceneBlob.size)}`}
+            </Text>
+            <Button size="mds" onClick={() => setSceneBlob(undefined)} fill="outline">
+              Clear Scene
+            </Button>
+          </>
+        ) : (
+          <>
+            <Text variant="b3">Upload gltf world scene file.</Text>
+            <Button
+              size="mds"
+              onClick={async () => {
+                const data = await platform.openFile("*");
```

Comment:
I am not sure about the mime type for scene file so add *

Reply:
https://www.khronos.org/registry/glTF/specs/2.0/glTF-2.0.html#glb-file-format-specification-file-extension


It should be `model/gltf-binary` and `.glb`. I think you'll want both the mimetype and the extension.